File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  let reversed = ""
  for(let i = str.length - 1; i > -1; i--) {
    reversed = reversed + str[i]
  }
  return reversed
}

if (require.main === module) {
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// pseudocode
// Initialize a variable called reversed with an empty string
// iterate over the string backwards 
  // and add each character to the end of reversed
// return reversed

// explanation
// Initialize a variable called reversed with an empty string
// iterate over the string backwards and add each character to the end of reversed

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  reversed = ""
  str.chars.each do |char|
    reversed = char + reversed
  end
  reversed
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')
end

# Please add your pseudocode to this file
# initialize reversed as an empty string
# iterate over each character in the string:
  # input the character in front of the previous characters and store in reversed
# return reversed

# And a written explanation of your solution
# initialize a variable as an empty string. 
# iterate over the string, adding each character in front
# of the previous character.